shared/process-tc-artifacts.sh

```diff
@@ -47,8 +64,56 @@ tar -x -z -C generated-$PLATFORM -f $PLATFORM.generated-files.tar.gz
 
 date
 
+# Normalize the rust save-analysis files so that instead of having absolute
+# paths they have searchfox-normalized paths where source paths have no prefix
+# and objdir paths start with __GENERATED__.
+
+# Rust stdlib looks like either "src/libstd/num.rs" on linux or
+# "src\\libstd\\num.rs" on windows.  We normalize to __GENERATED__/__RUST__.
+# We do this normalization first because there's no "src" directory at the root
+# of mozilla-central, so we can safely assume that if we have a relative path
+# that starts with "src/" that it's rust code.  Hopefully.  (If we did it later,
+# we'd be seeing the results of other absolute path normalizations.)
+NORMALIZE_EXPR='s#"src[/\\]#"__GENERATED__/__RUST__/#gI'
+# For some reason we see generated paths under checkouts like:
+# /builds/worker/checkouts/gecko/obj-arm-unknown-linux-androideabi/dist/xpcrs/rt/nsIChannel.rs
+# Handle this, and do it before we do the source normalization in the next line.
+NORMALIZE_EXPR+=';s#/builds/worker/checkouts/gecko/obj-[-a-zA-Z0-9_]+/#__GENERATED__/#g'
+# Non-windows source paths get normalized off.
+NORMALIZE_EXPR+=';s#/builds/worker/checkouts/gecko/##g'
+# Non-windows build paths get normalized to __GENERATED__.
+NORMALIZE_EXPR+=';s#/builds/worker/workspace/obj-build/#__GENERATED__/#g'
+# I haven't actually seen the same weird source path with "obj-*" under it on
+# Windows yet, but let's proactively assume such a thing will happen and add
+# the following regexp just in case.  But right now nsIChalle.rs on Windows
```

Comment:
nsIChalle.rs presumably should be nsIChannel.rs?

Reply:
Yes, thanks.  I'm really hoping this was a result of me switching back to CUA bindings in VS Code somehow!